### DISCUSSION TITLE
Graph-based simplification of quantum circuits via ZX-calculus

### DISCUSSION THREAD

**Discussion by Fe-r-oz** [QuantumSavory/QuantumClifford.jl#432] [Ideas]
Hi, Stefan!

The paper [Graph-theoretic Simplification of Quantum Circuits with the ZX-calculus](https://arxiv.org/pdf/1902.03178) suggests that Clifford and Clifford+T circuits can be simplified via ZX-Calculus. The Python tool from this paper: https://github.com/zxcalc/pyzx

Examples of [clifford simplification](https://yaoquantum.org/ZXCalculus.jl/dev/examples/#Clifford-simplification) from [ZXCalculus.jl](https://github.com/QuantumBFS/ZXCalculus.jl) inspired from [pyxz](https://github.com/zxcalc/pyzx). 

What are your thoughts on `QuantumCliffordZXCalculusExt` module? Does this fit into the bigger picture?

**Comment by Krastanov** [comment]
It fits in the bigger picture, but any extensions should just wrap around external ZXcalculus libraries, not develop anything as part of this library. And we first need a usecase, before we work on such extensions.

**Reply by Fe-r-oz** [reply]
I think the discussion is relevant in the bigger picture, so reopening the discussion. Since my query was answered, I thought to close the discussion 😅

**Reply by Fe-r-oz** [reply]
This [wonderful thesis](https://arxiv.org/pdf/2501.17959) posted on 29th Jan, 2025 might presents the first use-case. Portion of Abstract is very interesting so I attached it 😅 

**Partial Abstract**: _While stabilizer tableaus have proven exceptionally useful as a descriptive tool for additive quantum codes, they otherwise offer little guidance for concrete constructions or coding algorithm analysis 🗿 . We introduce a representation of stabilizer codes as graphs with certain structures. Specifically, the graphs take a semi-bipartite form wherein input nodes map to output nodes, such that output nodes may connect to each other but input nodes may not. Intuitively, the graph’s input-output edges represent information propagation of the encoding circuit, while output-output edges represent the code’s entanglement structure. We prove that this graph representation is in bijection with tableaus and give an efficient compilation algorithm that transforms tableaus into graphs. We then show that this map is efficiently invertible, which gives a new universal recipe for code construction by way of finding graphs with sufficiently nice properties...._ 

The use-case is of ZX-Calculus for this graph-based formalism formalism, not of graph-based simplification of quantum circuits via ZX-calculus.
